File: Corte 1/Reto Biblioteca/biblioteca.py
# Listas con los nombres de los recursos y los días de la semana
recursos = ["Computador", "Videobeam", "Sala"]
dias = ["Lunes", "Martes", "Miercoles", "Jueves", "Viernes"]

# Matriz que contiene los préstamos de cada recurso (filas) para cada día (columnas)
matriz = [
    [4, 6, 2, 8, 5],
    [2, 3, 1, 4, 2],
    [5, 5, 4, 6, 8]
]

# Se obtienen las dimensiones de la matriz (3 filas x 5 columnas)
numFilas = len(matriz)
numColumnas = len(matriz[0])

# Construir el encabezado alineado con los nombres de días y la columna 'Total'
titulos = f"{'Recurso':<12}" + "".join([f"{dias[j]:>12}" for j in range(numColumnas)]) + f"{'Total':>12}"
print(titulos)
print("-" * len(titulos)) # Imprimir línea divisoria

# Recorrer cada fila (recurso) de la matriz
for i in range(numFilas):
    # Calcular la suma total del recurso actual usando la función sum() en la fila
    sumaRecurso = sum(matriz[i])
    
    # Formatear la fila actual con los datos del recurso, préstamos diarios y el total acumulado
    fila = f"{recursos[i]:<12}" + "".join([f"{matriz[i][j]:>12}" for j in range(numColumnas)]) + f"{sumaRecurso:>12}"
    print(fila)

# Variables auxiliares para almacenar el valor máximo de préstamos y el nombre del día
maxCarga = -1
diaMasCargado = ""

# Recorrer la matriz por columnas (días de la semana)
for j in range(numColumnas):
    # Acumulador para sumar el total de préstamos del día actual
    sumDia = 0
    
    # Sumar los valores de cada recurso prestado en el día j
    for i in range(numFilas):
        sumDia += matriz[i][j]
    
    # Comparar la suma del día con el máximo actual y actualizar si es mayor
    if sumDia > maxCarga:
        maxCarga = sumDia
        diaMasCargado = dias[j]

# Imprimir el resultado final del día con mayor actividad
print("\n" + "-" * 40)
print(f"El dia mas cargado fue el {diaMasCargado} con un total de {maxCarga} prestamos.")

File: Corte 1/Reto Biblioteca/biblioteca.cpp
#include <iostream>

using namespace std;

int main() {
    // Arreglos para almacenar los nombres de los recursos y los días de la semana
    const char* recursos[3] = {"Computador", "Videobeam ", "Sala      "};
    const char* dias[5] = {"Lunes", "Martes", "Miercoles", "Jueves", "Viernes"};

    // Matriz de 3 filas (recursos) por 5 columnas (días) con los datos de préstamos
    int matriz[3][5] = {
        {4, 6, 2, 8, 5},
        {2, 3, 1, 4, 2},
        {5, 5, 4, 6, 8}
    };

    // Imprimir el encabezado de la tabla formateado con tabulaciones
    cout << "Recurso   \tLunes\tMartes\tMiercoles\tJueves\tViernes\tTotal\n";
    cout << "-----------------------------------------------------------------------\n";

    // Recorrer las filas de la matriz (recursos)
    for (int i = 0; i < 3; i++) {
        // Imprimir el nombre del recurso actual
        cout << recursos[i] << "\t";
        
        // Variable acumuladora para el uso total del recurso actual
        int sumaRecurso = 0;
        
        // Recorrer las columnas (días) del recurso actual
        for (int j = 0; j < 5; j++) {
            cout << matriz[i][j] << "\t";
            sumaRecurso += matriz[i][j]; // Sumar los préstamos del día al total del recurso
        }
        
        // Imprimir el total calculado para el recurso y salto de línea
        cout << sumaRecurso << "\n";
    }

    // Variables para rastrear el día con la mayor suma de préstamos
    int maxCarga = -1;
    const char* diaMasCargado = "";

    // Recorrer la matriz por columnas (días) para evaluar la carga de cada día
    for (int j = 0; j < 5; j++) {
        // Variable acumuladora para el total de préstamos del día actual
        int sumaDia = 0;
        
        // Sumar todos los recursos prestados en el día j
        for (int i = 0; i < 3; i++) {
            sumaDia += matriz[i][j];
        }
        
        // Actualizar el valor máximo si el día actual supera al mayor registrado
        if (sumaDia > maxCarga) {
            maxCarga = sumaDia;
            diaMasCargado = dias[j];
        }
    }

    // Imprimir los resultados del día más cargado
    cout << "\n-----------------------------------\n";
    cout << "El dia mas cargado fue el " << diaMasCargado 
         << " con un total de " << maxCarga << " prestamos.\n";

    return 0;
}